src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ImplicitConversion.cs

```diff
@@ -165,11 +163,10 @@ public bool Bind()
                 FUNDTYPE ftDest = _typeDest.fundType();
                 Debug.Assert(ftDest != FUNDTYPE.FT_NONE || _typeDest is ParameterModifierType);
 
+                Debug.Assert(Enum.IsDefined(typeof(TypeKind), _typeSrc.GetTypeKind()));
+                Debug.Assert(_typeSrc.GetTypeKind() != TypeKind.TK_NullableType);
                 switch (_typeSrc.GetTypeKind())
                 {
-                    default:
```

Comment:
This would protect against adding new type kinds without redesigning this switch statement.